sdcm/provision/gce/provisioner.py

```diff
@@ -0,0 +1,140 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2022 ScyllaDB
+from __future__ import annotations
+import datetime
+import logging
+import random
+from typing import Dict, List
+
+from libcloud.compute.base import Node, NodeLocation
+
+from sdcm.keystore import KeyStore
+from sdcm.provision.gce import GCE_REGIONS
+from sdcm.provision.gce.disk_struct_provider import DiskStructProvider, DiskStructArgs
+from sdcm.provision.gce.metadata_provider import MetadataProvider, MetadataArgs
+from sdcm.provision.gce.virtual_machine_provider import VirtualMachineProvider
+from sdcm.provision.provisioner import Provisioner, InstanceDefinition, VmInstance, PricingModel
+from sdcm.test_config import TestConfig
+from sdcm.utils.gce_utils import get_gce_service
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GCEProvisioner(Provisioner):
+    """Provides API for VM provisioning in GCE"""
+
+    def __init__(self, test_id, region, params: dict):
```

Comment:
It would be very beneficial to have `gce_service` as argument, so we can override it in unit test with some fake implementation.

Reply:
I'll think about that, though I do like the simpler signature here :)

Reply:
you'll enjoy simple unit-testing better :)